landez/tiles.py

```diff
@@ -311,13 +329,33 @@ def run(self, force=False):
 
         # Go through whole list of tiles and gather them in tmp_dir
         self.rendered = 0
+
+        threadlist = []
+
         for (z, x, y) in tileslist:
-            try:
-                self._gather((z, x, y))
-            except Exception as e:
-                logger.warn(e)
-                if not self.ignore_errors:
-                    raise
+            t = threading.Thread(None, self._gather, None, ((z, x, y), ))
+            t.setDaemon(True)
+            t.start()
+            threadlist.append(t)
+
+        for t in threadlist:
+            if t.isAlive():
+                t.join()
+            if self.error:
+                break
+
+        if not self.ignore_errors and self.exception is not None:
+            raise self.exception
+
+        if len(self.grid_fields) > 0:
+            for (z, x, y) in tileslist:
+                t = threading.Thread(None, self._gather_grid, None, ((z, x, y), ))
+                t.setDaemon(True)
```

Comment:
I misunderstood what exactly the daemon attribute did.
I wanted to quit the others threads if one of them encountered an error (except if ignore_error was set) but the daemon attribute isn't a good way to do it so I changed it.

Anyway, thanks for the comments, if you see another error /bad practice, I'd be grateful if you report it.